apps/mobile/src/screens/auth/components/sign-up-form/sign-up-form.tsx

```diff
@@ -13,42 +24,97 @@ type Properties = {
 };
 
 const SignUpForm: React.FC<Properties> = ({ onSubmit }) => {
-	const { control, errors, handleSubmit } = useAppForm<UserSignUpRequestDto>({
-		defaultValues: USER_SIGN_UP_DEFAULT_VALUES,
-		validationSchema: userSignUpValidationSchema,
-	});
+	const [isPasswordHidden, setIsPasswordHidden] = useState<boolean>(true);
+	const [isConfirmPasswordHidden, setIsConfirmPasswordHidden] =
+		useState<boolean>(true);
+
+	const handlePasswordIconPress = (): void => {
+		setIsPasswordHidden(!isPasswordHidden);
+	};
+
+	const handleConfirmPasswordIconPress = (): void => {
+		setIsConfirmPasswordHidden(!isConfirmPasswordHidden);
+	};
+
+	const { control, errors, handleSubmit, setError } =
+		useAppForm<UserSignUpFormDto>({
+			defaultValues: USER_SIGN_UP_DEFAULT_VALUES,
+			validationSchema: userSignUpValidationSchema,
+		});
 
 	const handleFormSubmit = useCallback((): void => {
-		void handleSubmit(onSubmit)();
-	}, [handleSubmit, onSubmit]);
+		void handleSubmit((data: UserSignUpFormDto) => {
+			const { confirmPassword, ...userData } = data;
+
+			if (confirmPassword === userData.password) {
+				onSubmit(userData);
+			} else {
+				setError(UserCustomValidation.FIELD, {
+					message: UserCustomValidation.ERROR_MESSAGE,
+					type: UserCustomValidation.ERROR_TYPE,
+				});
+			}
+		})();
+	}, [handleSubmit, onSubmit, setError]);
 
 	return (
-		<View>
-			<Text>Sign Up</Text>
+		<>
+			<Text preset="uppercase" size="xl" weight="bold">
+				CREATE AN ACCOUNT
+			</Text>
+			<Text style={globalStyles.mb16}>
+				Already have an account? Go to{" "}
+				<Link label="Sign In" to={`/${RootScreenName.SIGN_IN}`} />
+			</Text>
 			<Input
 				control={control}
 				errors={errors}
-				label="Email"
-				name="email"
-				placeholder="Enter your email"
+				isAutoFocused
+				label="Name"
+				name="name"
+				placeholder="name"
 			/>
 			<Input
 				control={control}
 				errors={errors}
-				label="Name"
-				name="name"
-				placeholder="Enter your name"
+				label="Email"
+				name="email"
+				placeholder="name@gmail.com"
 			/>
 			<Input
+				accessoryRight={
+					<IconButton
+						iconColor={BaseColor.LIGHT_GRAY}
+						iconSize={20}
+						name={isPasswordHidden ? "visibility-off" : "visibility"}
```

Comment:
Let's move this ternary to a separate little function inside the SignUp component. We will be able to reuse

```
const getInputVisibility: () => IconName = () => isPasswordHidden ? "visibility-off" : "visibility"
```